utilities/test_suite/HOST/Tensor_audio_host.cpp

```diff
@@ -361,45 +368,16 @@ int main(int argc, char **argv)
                     RpptMelScaleFormula melFormula = RpptMelScaleFormula::SLANEY;
                     Rpp32s numFilter = 80;
                     bool normalize = true;
-                    Rpp32s srcDimsTensor[] = {257, 225, 257, 211, 257, 214}; // (height, width) for each tensor in a batch for given QA inputs.
-                    // Accepts outputs from FT layout of Spectrogram for QA
-                    srcDescPtr->layout = dstDescPtr->layout = RpptLayout::NFT;
-
-                    maxDstHeight = 0;
-                    maxDstWidth = 0;
-                    maxSrcHeight = 0;
-                    maxSrcWidth = 0;
-                    for(int i = 0, j = 0; i < batchSize; i++, j += 2)
-                    {
-                        maxSrcHeight = std::max(maxSrcHeight, (int)srcDimsTensor[j]);
-                        maxSrcWidth = std::max(maxSrcWidth, (int)srcDimsTensor[j + 1]);
-                        dstDims[i].height = numFilter;
-                        dstDims[i].width = srcDimsTensor[j + 1];
-                        maxDstHeight = std::max(maxDstHeight, (int)dstDims[i].height);
-                        maxDstWidth = std::max(maxDstWidth, (int)dstDims[i].width);
-                    }
-
-                    srcDescPtr->h = maxSrcHeight;
-                    srcDescPtr->w = maxSrcWidth;
-                    dstDescPtr->h = maxDstHeight;
-                    dstDescPtr->w = maxDstWidth;
-
-                    set_audio_descriptor_dims_and_strides_nostriding(srcDescPtr, batchSize, maxSrcHeight, maxSrcWidth, maxSrcChannels, offsetInBytes);
-                    set_audio_descriptor_dims_and_strides_nostriding(dstDescPtr, batchSize, maxDstHeight, maxDstWidth, maxDstChannels, offsetInBytes);
-                    srcDescPtr->numDims = 3;
-                    dstDescPtr->numDims = 3;
-
-                    srcDescPtr->numDims = 3;
-                    dstDescPtr->numDims = 3;
-
-                    // Set buffer sizes for src/dst
-                    unsigned long long spectrogramBufferSize = (unsigned long long)srcDescPtr->h * (unsigned long long)srcDescPtr->w * (unsigned long long)srcDescPtr->c * (unsigned long long)srcDescPtr->n;
-                    unsigned long long melFilterBufferSize = (unsigned long long)dstDescPtr->h * (unsigned long long)dstDescPtr->w * (unsigned long long)dstDescPtr->c * (unsigned long long)dstDescPtr->n;
-                    inputf32 = (Rpp32f *)realloc(inputf32, spectrogramBufferSize * sizeof(Rpp32f));
-                    outputf32 = (Rpp32f *)realloc(outputf32, melFilterBufferSize * sizeof(Rpp32f));
-
-                    // Read source data
-                    read_from_bin_file(inputf32, srcDescPtr, srcDimsTensor, "spectrogram", scriptPath);
+                    Rpp32s srcDimsTensor[batchSize * 2];
+                    // (height, width) for each tensor in a batch for given QA inputs.
+                    srcDimsTensor[0] = 257;
```

Comment:
why is this changed to single assignments

Reply:
changed on host, doing it on hip may involve an additional hipmemcpy